textdb/textdb-textql/src/main/javacc/TextQLParser.jj

```diff
@@ -0,0 +1,311 @@
+{
+    String identif;
+    List<String> identifiers = new ArrayList<String>();
+}
+{
+    identif=identifier() { identifiers.add(identif); }
+    ( <COMMA> identif=identifier() { identifiers.add(identif); } )*
+    { return identifiers; }
+}
+
+
+/*
+ *	Statements Declaration
+ */
+List<Map<String, Object>> statementsMain(Consumer<Map<String,Object>> statementConsumer) : 
+{
+    List<Map<String, Object>> statements = new ArrayList();
+    Map<String, Object> newStatement;
+}
+{
+    (
+        newStatement = statement() { statements.add(newStatement); }
+        { if(statementConsumer!=null)statementConsumer.accept(newStatement); }
+    )*
+    <EOF>
+    { return statements; }
+}
+
+Map<String, Object> statement() : 
+{
+    Map<String, Object> newStatement;
+}
+{
+    (
+            newStatement = selectStatement() 
+        |
+            newStatement = createViewStatement()
+        |
+            newStatement = emptyStatement()
+    )
+    <SEMICOLON>
+    { return newStatement; }
+}
+
+Map<String, Object> selectStatement() : 
+{
+    HashMap<String, Object> statementParameters = new HashMap<String, Object>() {{
+        put("statementType", "select");
+        put("statementName", getNewUniqueId());
+        put("projectAll", null);
+        put("projectFields", null);
+        put("extractCommand", null);
+        put("from", null);
+        put("limit", null);
+        put("offset", null);
+    }};
+    
+    List<String> projectFields;
+    Map<String,Object> extractCmd;
+    String identifier;
+    String from;
+    Integer limit;
+    Integer offset;
+
+}
+{
+    (
+            <SELECT> 
+            ( 
+                    <STAR> { statementParameters.put("projectAll", true);  } 
+                | 
+                    { projectFields = new ArrayList<String>(); }
+                      identifier=identifier() { projectFields.add(identifier); }
+                    ( <COMMA> identifier=identifier() { projectFields.add(identifier); } )*
+                    { statementParameters.put("projectFields", projectFields); }
+            )
+            ( <EXTRACT> extractCmd=extractCommand() { statementParameters.put("extractCommand", extractCmd); } )?
+        |
+            
+            <EXTRACT> extractCmd=extractCommand() { statementParameters.put("extractCommand", extractCmd); } 
+    )
+    <FROM> from=identifier() { statementParameters.put("from", from); }
+    ( <LIMIT> limit=numberInteger() { statementParameters.put("limit", limit); })?
+    ( <OFFSET> offset=numberInteger() { statementParameters.put("offset", offset); } )?
+    { return statementParameters; }
+}
+
+Map<String, Object> createViewStatement() : 
+{
+    HashMap<String, Object> statementParameters = new HashMap<String, Object>() {{
+        put("statementType", "view");
+        put("statementName", null);
+        put("substatement", null);
+    }};
+    String viewName;
+    Map<String, Object> substatement;
+}
+{
+    <CREATE> 
+    <VIEW>
```

Comment:
This shouldn't need to be a single literal. It is very natural for the literals to have common prefixes.